scipy/stats/tests/test_stats.py

```diff
@@ -4080,6 +4041,235 @@ def test_ttest_ind_permutation_check_inputs(self):
+        res_vec_list = [[s, p] for (s, p) in
+                        zip(res_vec.statistic, res_vec.pvalue)]
+
+        # move axis so that samples are 1 per row for easy comparison
+        # individually
+        a_split = np.moveaxis(a, 0, 1)
+        b_split = np.moveaxis(b, 0, 1)
+        # compute the results individually comparing sample 1 from `a` to
+        # sample 1 in `b` and so on.
+        res_individual = [(stats.ttest_ind(a, b, trim=.20, equal_var=False))
+                          for (a, b) in zip(a_split, b_split)]
+
+        for (i, v) in zip(res_individual, res_vec_list):
+            assert_array_equal(i, v)
+
+    @pytest.mark.parametrize('alt,pr,tr',
+                             [['greater', 0.9985605452443, -4.2461168970325],
+                              ['less', 0.001439454755672, -4.2461168970325]])
+    def test_alternatives(self, alt, pr, tr):
+        '''
+        > library(PairedData)
+        > a <- c(2.7,2.7,1.1,3.0,1.9,3.0,3.8,3.8,0.3,1.9,1.9)
+        > b <- c(6.5,5.4,8.1,3.5,0.5,3.8,6.8,4.9,9.5,6.2,4.1)
+        > options(digits=16)
+        > yuen.t.test(a, b, alternative = 'greater')
+        '''
+        a = [2.7, 2.7, 1.1, 3.0, 1.9, 3.0, 3.8, 3.8, 0.3, 1.9, 1.9]
+        b = [6.5, 5.4, 8.1, 3.5, 0.5, 3.8, 6.8, 4.9, 9.5, 6.2, 4.1]
+
+        t, p = stats.ttest_ind(a, b, trim=.2, equal_var=False, alternative=alt)
+        assert_allclose(p, pr, atol=1e-10)
+        assert_allclose(t, tr, atol=1e-10)
+
+    def test_errors_unsupported(self):
+        # confirm that attempting to trim with NaNs or permutations raises an
+        # error
+        with assert_raises(ValueError, match="Permutations are currently not "
+                                             "supported with trimming."):
+            stats.ttest_ind([1, 2], [2, 3], trim=.2, permutations=2)
+
+        with assert_raises(ValueError, match="nan-containing/masked inputs "
+                                             "with nan_policy='omit' are "
+                                             "currently not supported by "
+                                             "permutation tests, one-sided "
+                                             "asymptotic tests, or trimmed "
+                                             "tests."):
+            stats.ttest_ind([1, 2], [2, np.nan, 3], trim=.2, nan_policy='omit')
+
+    def test_nans_on_axis(self):
+        # confirm that with `nan_policy='propagate'`, NaN results are returned
+        # on the correct location
+        a = np.random.randint(10, size=(5, 3, 10)).astype('float')
+        b = np.random.randint(10, size=(5, 3, 10)).astype('float')
+        # set some indices in `a` and `b` to be `np.nan`.
+        a[0][2][3] = np.nan
+        b[2][0][6] = np.nan
+
+        # suppress a warning from ttest_finish, as a result of some results in
+        # the output being NaN. Warning unrelated to trimming.
+        with suppress_warnings() as sup, np.errstate(invalid="ignore"):
```

Comment:
OK. If it were related, I would suggest using `with np.errstate...` wherever it actually occurs. I don't think there needs to be a warning when there are nans with `nan_policy='propagate'`.

Reply:
I see. It looks like the old build that this failure was from has been deleted/archived, so I will have to push another commit to see the stacktrace.

Reply:
Here's the stacktrace (variables ommitted): [link](https://gist.github.com/4af7edc9de3142e8449ff728164eb152) 
```
2021-04-11T22:56:28.6050909Z scipy/stats/tests/test_stats.py:4227: in test_nans_on_axis
2021-04-11T22:56:28.6051791Z     res = stats.ttest_ind(a, b, trim=.2, axis=-1)
...
2021-04-11T22:56:28.6057843Z scipy/stats/stats.py:6180: in ttest_ind
2021-04-11T22:56:28.6058507Z     res = _ttest_ind_from_stats(m1, m2, denom, df, alternative)
...
2021-04-11T22:56:28.6090337Z scipy/stats/stats.py:5724: in _ttest_ind_from_stats
2021-04-11T22:56:28.6090967Z     t, prob = _ttest_finish(df, t, alternative)
...
2021-04-11T22:56:28.6110396Z scipy/stats/stats.py:5708: in _ttest_finish
2021-04-11T22:56:28.6110900Z     prob = 2 * distributions.t.sf(np.abs(t), df)
...
2021-04-11T22:56:28.6116289Z scipy/stats/_distn_infrastructure.py:2036: in sf
2021-04-11T22:56:28.6116792Z     cond2 = cond0 & (x <= _a)
2021-04-11T22:56:28.6117299Z E   RuntimeWarning: invalid value encountered in less_equal
```

This result makes me think that it occurs whether trimming happens or not, but it is extremely hard to test as it doesn't happen on my test system. 

For now I will suppress it in `ttest_ind`.